0.27rc1 proposed changes to GUI and throughput

mdcato opened the pull request:
I propose some changes to 0.27rc1.
I saw an anomaly in throughput values in GUI's final summary screen. Some drives, and a different number of them on different runs, would show a throughput of 1 B/s. I have added code to capture start/end times of runs, and report "Overall" throughput on final summary. This also is in response to long-standing throughput values being a "most recent" value which was not a good representation of the drive's overall performance. Along with this, there are added nwipe_log entries in method.c to mark the start & end.
In gui.c I have changed the running and final summary model/serial#/size display to show these in a more columnar format rather than having the fields smashed up against each other which makes your eyes have to hunt for the values when several drives are being wiped simultaneously.
I have changed the "[syncing]" field to come after the rest of the values so it does not "push" the throughput value to the right.
Original:
![image](https://user-images.githubusercontent.com/55356734/73997005-e9e47280-4922-11ea-9d56-90a540c9ba59.png)

This Pull Request:
![image](https://user-images.githubusercontent.com/55356734/73997026-f5d03480-4922-11ea-8a5d-88cfe6a2523e.png)

Original summary:
![image](https://user-images.githubusercontent.com/55356734/73997036-ff599c80-4922-11ea-91e4-3eb7b08987cd.png)

This Pull Request summary; note "Overall" is calculated from start/stop times of runs, one of the "most recent" throughput values is "1 B/s":
![image](https://user-images.githubusercontent.com/55356734/73997049-084a6e00-4923-11ea-9da2-53141f7bd7f3.png)

Changes were made in gui.c to select string values within a switch statement and then doing a single display with wprintw(). This is a baby step toward multi-lingual support (expand list of strings to a table where columns represent different languages) and makes it a single place to change on-screen formatting. (Again, I like to columnarize things to make it easy to visually scan for info.)
In gui.c, I made a "political" change in line 1677 from "American Department of Defense" to "U.S. Department of Defense". While the USA is colloquially called "America" the DoD's official name is "U.S. Department of Defense", and other countries in the Americas might not like being lumped in with the U.S.'s DoD.
Changes were made in context.h, device.c and gui.c to add a "device_size_GB" which calculates the device's size once in GiB for later display. A minor efficiency improvement since there are very few sub-1GiB devices floating around anymore, and handles TB nicely (for now).

PartialVolume commented:
@mdcato looks like you need to run make format after make so the CI style checks pass before submitting the pull request. This will need to be done before I can commit. Overall I like the changes you've made.

mdcato commented:
Yeah, I probably made a couple of changes after the last format.


(sent from mobile)

________________________________
From: PartialVolume <notifications@github.com>
Sent: Thursday, February 6, 2020 9:42:41 PM
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@mdcato<https://github.com/mdcato> looks like you need to run make format after make so the CI style checks pass before submitting the pull request. This will need to be done before I can commit. Overall I like the changes you've made.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PQLBTTT6AILBR4JFUDRBTKDDA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELBTZUY#issuecomment-583220435>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PW7FG6YCGPKRKI6KETRBTKDDANCNFSM4KRHR44Q>.

mdcato commented:
Hmmm, I ran “make format” and uploaded the files that it changed, but am still getting two errors.  Unfortunately it doesn’t indicate WHAT the errors are, or I don’t know where to look.  A difference in “make format” and the workflow?  A little help, please?
[cid:image001.jpg@01D5DD3D.28D74DC0]

From: PartialVolume <notifications@github.com>
Sent: Thursday, February 6, 2020 21:43
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@mdcato<https://github.com/mdcato> looks like you need to run make format after make so the CI style checks pass before submitting the pull request. This will need to be done before I can commit. Overall I like the changes you've made.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PQLBTTT6AILBR4JFUDRBTKDDA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELBTZUY#issuecomment-583220435>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PW7FG6YCGPKRKI6KETRBTKDDANCNFSM4KRHR44Q>.

PartialVolume commented:
@louib are you able to help with what might be causing style checks to fail for @mdcato's PR? 

@mdcato just to get some basic background info, what distro & version are you compiling on and what version of clang-tidy. Probably nothing to do with it but just in case there was some version issue with clang.

@louib I don't know if this is possible but can the CI check be made more verbose so we can see which file it's failing on ?

mdcato commented:
Fedora Core 31
clang-format 9.0.0
gcc 9.2.1

________________________________
From: PartialVolume <notifications@github.com>
Sent: Thursday, February 6, 2020 10:46 PM
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@louib<https://github.com/louib> are you able to help with what might be causing style checks to fail for @mdcato<https://github.com/mdcato>'s PR?

@mdcato<https://github.com/mdcato> just to get some basic background info, what distro & version are you compiling on and what version of clang-tidy. Probably nothing to do with it but just in case there was some version issue with clang.

@louib<https://github.com/louib> I don't know if this is possible but can the CI check be made more verbose so we can see which file it's failing on ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PV262J6Y3T5HA7KO4LRBTRRLA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELBWUHY#issuecomment-583232031>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PW7TD7CGPKFVUZBEMTRBTRRLANCNFSM4KRHR44Q>.

louib commented:
I'm running `clang-format` version 7 and this is what I get when running `clang-format -style=file -i src/*.c src/*.h`:
```
--- a/src/nwipe.c
+++ b/src/nwipe.c
@@ -573,7 +573,8 @@ void* signal_hand( void* ptr )
         {
 
             // Log current status. All values are automatically updated by the GUI
-            case SIGUSR1: {
+            case SIGUSR1:
+            {
                 compute_stats( ptr );
 
                 for( i = 0; i < nwipe_misc_thread_data->nwipe_selected; i++ )
@@ -642,7 +643,8 @@ void* signal_hand( void* ptr )
             case SIGHUP:
             case SIGINT:
             case SIGQUIT:
-            case SIGTERM: {
+            case SIGTERM:
+            {
                 /* Set termination flag for main() which will do housekeeping prior to exit */
                 terminate_signal = 1;
 
```

It looks like the `AfterCaseLabel` style config was added in a later version of `clang-format`, which is why were having problem.

PartialVolume commented:
@louib what's our options ? I can download the PR branch and make the changes to the PR to deal with this. I tend to download each PR branch anyway so not really much extra work for me. Is there any other way of handling this situation ?

mdcato commented:
I modified the .clang_format config file to add AfterCaseLabel: false, but only nwipe.c was affected, so there is still an additional formatting option affecting a different file (two incidences are reported when I uploaded the PR). It would be helpful if the work flow would show which files/lines were being flagged.


(sent from mobile)

________________________________
From: PartialVolume <notifications@github.com>
Sent: Saturday, February 8, 2020 7:36:09 AM
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@louib<https://github.com/louib> what's our options ? I can download the PR branch and make the changes to the PR to deal with this. I tend to download each PR branch anyway so not really much extra work for me. Is there any other way of handling this situation ?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PXGLQTRMVEJPKGQFBTRB2YMTA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELFR64Y#issuecomment-583737203>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PWOD7GVEYU7U3K5VPDRB2YMTANCNFSM4KRHR44Q>.

louib commented:
@mdcato you should see the formatting diff in the CI output now!

mdcato commented:
Mistakenly put in top folder, deleted, then updated to src folder. Sorry for the girations.


(sent from mobile)

________________________________
From: PartialVolume <notifications@github.com>
Sent: Saturday, February 8, 2020 12:23:34 PM
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@PartialVolume requested changes on this pull request.

I'm not sure what's going on here with nwipe.c. It looks like you may have moved nwipe.c to the root folder then done a commit then realised the mistake then copied nwipe.c back and deleted it from the root folder? Makes it tricky to check for changes in nwipe.c, I'd need to manually diff against current version to see any changes. Makes the history a bit messy.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PRJSWDURIG4XFD2HH3RB32CNA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCUYW7RA#pullrequestreview-355561412>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PRWYMXNSWIXG2MPABDRB32CNANCNFSM4KRHR44Q>.

louib commented:
@mdcato could you squash the commits so it's easier to review?

mdcato commented:
The short answer is I can learn how to squash, but it won’t be done quickly.  Looks like it must be done from command line unless I overlooked something, and I haven’t been using command line so will take a while to gain confidence in that method.
The long answer is only the first commit contained code changes to context.h, device.c, gui.c, and method.c; the subsequent ones were attempts to configure clang_format 9.0.0 to be agreeable with the workflow checking.  I did not change nwipe.c, it only popped up because of the clang_format version differences.
I’m happy to help resolve clang_format differences if I’m the only one running Fedora 31 with clang_format 9.0.0 as this makes the nwipe project more robust.
You said earlier “you should see the formatting diff in the CI output now!”, but that seemingly requires another commit which circles back to the squash issue.  Following the link from the “Run failed for” email just shows “X verifying code style”.

From: louib <notifications@github.com>
Sent: Saturday, February 8, 2020 12:28
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@mdcato<https://github.com/mdcato> could you squash the commits so it's easier to review?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PTJRIMWC4CMIAXDJNDRB32UTA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELFYOZY#issuecomment-583763815>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PUFW4FR76BN7I6DL6TRB32UTANCNFSM4KRHR44Q>.

louib commented:
@PartialVolume maybe you can do the squashing, can you push directly on the PR's branch? I'm pretty sure I can't

PartialVolume commented:
The way I tend to do it, is do a git reset --hard HEAD@{2} the number depends on how far you want to go back. I then check the changes are the one I want to commit. I then stash those changes. Create a new branch from the master (assuming you were working in a branch and not your master). Switch to that branch. Then pop the stash. Do the commit. Check the history looks good. I'll download your branch and run through the commands I used and post them here.

PartialVolume commented:
@mdcato I downloaded your branch to run through the commands I would use to get back to the commit prior to where you had nwipe.c in the root folder. This assumes your working on a PC and pushing to your Github fork.

**Summary**
First delete your branch mdcato-nwipe-0.27rc1-patch-1 on github. This needs to be done because that branch will be two or three commits ahead and will want you to pull before you push which isn't what we want.

Then on the command line on the root folder of nwipe

> git branch (shows brances)
> git checkout mdcato-nwipe-0.27rc1-patch-1
> git log
> git reset --hard 29b3455f8cac9b73d793ca38f0e8a4b14156cd47
> vi .clang-format (Make the changes ie add AfterCaseLabel:  false)
> git status 
> git add .clang-format
> git status
> git commit -m "Updated .clang_format config to add AfterCaseLabel: false"
> git push origin mdcato-nwipe-0.27rc1-patch-1
  
  Delete the existing pull request and submit a new pull request in github.

If you need a more detailed explanation of any of the steps let me know.

Here's what the history looks like after I made the changes and nwipe.c in now correctly in /src rather than root folder.
`commit 3256aa07d71b7730ab95ac40f936cedb1bdfb3eb (HEAD -> mdcato-nwipe-0.27rc1-patch-1)
Author: PartialVolume <nick.craig.law@gmail.com>
Date:   Sat Feb 8 21:37:00 2020 +0000

    Updated .clang_format config to add AfterCaseLabel: false

commit 29b3455f8cac9b73d793ca38f0e8a4b14156cd47
Author: mdcato <55356734+mdcato@users.noreply.github.com>
Date:   Thu Feb 6 22:05:54 2020 -0600

    after "make format"

commit 25f4c6f7c8f4168bcc24dcbbb8a084623ca96d08
Author: mdcato <55356734+mdcato@users.noreply.github.com>
Date:   Thu Feb 6 12:40:51 2020 -0600

    Add files via upload

commit 6d797b742cc7924becb0c912dcaa043bcc28df42 (origin/master, origin/HEAD, master)
Author: PartialVolume <nick.craig.law@gmail.com>
Date:   Tue Jan 21 23:29:01 2020 +0000

    Update CHANGELOG.md

commit 5c60dce0c5d9c9f2712227ff1e81d472343eb87b
Author: PartialVolume <nick.craig.law@gmail.com>
Date:   Tue Jan 21 23:25:33 2020 +0000

    Update CHANGELOG.md

commit 49b35aef8a0699490d3b29ab0bd1c13b005dd65d
Merge: ead9a78 a3a056c
Author: PartialVolume <nick.craig.law@gmail.com>
Date:   Mon Jan 13 09:49:05 2020 +0000

    Merge pull request #192 from louib/normalize_workflow_names
    
    Normalizing workflow names.

commit a3a056ce75ddbf3fb20f35a221883860bfd04168
Author: louib <L0U13@protonmail.com>
:`

PartialVolume commented:
In the above steps I used vi to make changes to .clang-format but you can of course using any text editor you like. I tend to use kate as I'm usually working in a KDE environment. But as I have a background on old sun unix systems i often just use vi. You may be working in a Integrated development environment (IDE) like kdevelop in which case you probably be editing inside that.

PartialVolume commented:
There is also a problem with this PR when used in a terminal I.e CNTRL-ALT-F2/F3 etc or shredOS (also 80 columns wide).

The device size is truncated as shown in the first two devices and wraps for the third device.

If columns are to be used they need to be full width justified based on whatever the detected width is so they work ok with a minimum width of 80 characters.
![15812843155336114552281954818195](https://user-images.githubusercontent.com/22084881/74110445-9a04e600-4b84-11ea-845d-728cbee76397.jpg)

mdcato commented:
That’s not what I see (refer to images in original Pull Request which while using a wider terminal, the lines do not exceed the 80-column width of the header window).  And verified using an 80-column terminal just now.  Difference in default font or ncurses implementation?  I’m using ncurses 6.1-12.20190803

Running screenshot from Fedora 31, 80-column terminal:
[cid:image001.png@01D5DF78.9293FFA0]

Summary screenshot:
[cid:image002.png@01D5DF78.9293FFA0]

From: PartialVolume <notifications@github.com>
Sent: Sunday, February 9, 2020 15:41
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


There is also a problem with this PR when used in a terminal I.e CNTRL-ALT-F2/F3 etc or shredOS (also 80 columns wide).

The device size is truncated as shown in the first two devices and wraps for the third device.

If columns are to be used they need to be full width justified based on whatever the detected width is so they work ok with a minimum width of 80 characters.
[15812843155336114552281954818195]<https://user-images.githubusercontent.com/22084881/74110445-9a04e600-4b84-11ea-845d-728cbee76397.jpg>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PQ55W4PWSZZGLWGF43RCBZ45A5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELGY6NQ#issuecomment-583896886>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PRKXGAZ7JC2A2MZA6TRCBZ45ANCNFSM4KRHR44Q>.

mdcato commented:
I also tried building shredOS so I could test on it, but the make fails because buildroot is an older version (from results of web search).  Fails building rpl_fseeko and freadahead.

From: Mike Cato / Hays Technical Services
Sent: Sunday, February 9, 2020 18:42
To: martijnvanbrummelen/nwipe <reply@reply.github.com>; martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mention <mention@noreply.github.com>
Subject: RE: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)

That’s not what I see (refer to images in original Pull Request which while using a wider terminal, the lines do not exceed the 80-column width of the header window).  And verified using an 80-column terminal just now.  Difference in default font or ncurses implementation?  I’m using ncurses 6.1-12.20190803

Running screenshot from Fedora 31, 80-column terminal:
[cid:image003.png@01D5DF82.8E04A7F0]

Summary screenshot:
[cid:image004.png@01D5DF82.8E04A7F0]

From: PartialVolume <notifications@github.com<mailto:notifications@github.com>>
Sent: Sunday, February 9, 2020 15:41
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com<mailto:nwipe@noreply.github.com>>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com<mailto:mike@haystech.com>>; Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


There is also a problem with this PR when used in a terminal I.e CNTRL-ALT-F2/F3 etc or shredOS (also 80 columns wide).

The device size is truncated as shown in the first two devices and wraps for the third device.

If columns are to be used they need to be full width justified based on whatever the detected width is so they work ok with a minimum width of 80 characters.
[15812843155336114552281954818195]<https://user-images.githubusercontent.com/22084881/74110445-9a04e600-4b84-11ea-845d-728cbee76397.jpg>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PQ55W4PWSZZGLWGF43RCBZ45A5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELGY6NQ#issuecomment-583896886>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PRKXGAZ7JC2A2MZA6TRCBZ45ANCNFSM4KRHR44Q>.

PartialVolume commented:
@mdcato I'm referring to before the wipe starts where there is slightly less columns due to the square brace selection [         ]  being present. The disc size is then lost.

After the wipe starts the braces aren't there so more room for the rest text and the size is available and you don't see the problem.

IDerr commented:
Hello, 
you take a look at this : https://github.com/IDerr/shredos
I took the old shreos package and repackaged it on a newer version of buildroot. Not tested yet, will do during this week, but it _should_ work

mdcato commented:
@PartialVolume, I see the problem, but the resolution requires a judgement call on what is the most important information.
The “simple” solution is to just limit the string length with the format specifier, for example “%25.25s” so that the resulting width is 25 and no more than 25 characters.
But for example, in your screenshot, “/dev/mapper/cryptswap1” is much longer than the allowed width for the pathname and ends in the model number’s space.  But if you have multiple devices with such a long device name, you’re probably interested in the right-most characters, but they are not likely to be drives that should be wiped.
I would also argue the serial number is more important than the model number, especially if you have multiple drives from the same mfr’s computer(s).
And then there are some mfr’s, like in my computer, that have the size built into the name: “Samsung SSD 970 EVO 250GB”, which is also why I initially chose “%25s” as the format specifier.
I’ll also note that the previous method of concatenating all the device info into a single string would also cut off the end.
So, keeping in mind it’s often the PFY that has to wrangle the drives, my stab at prioritizing fields this:

  1.  Size – often prominently displayed on drives label
  2.  Serial number – uniquely identifies drives, but is often more difficult to see on the label.  But if it *has* to be truncated, then the left-most characters should be lopped off as they are often the mfr’s initials (e.g. WD, PNY, etc) and the right-most ones uniquely identify the drive
  3.  Path – gives indication of interface (NVME, SATA); if truncated, I’d use left-most N characters, as the longer ones are (so far) like “/dev/mapper/cryptswap1” that wouldn’t be as likely a candidate for wiping
  4.  Model – It helps identify, but not as much as serial number
The resulting format string, which comes after the line’s first 13 chars where selection for wiping is shown, and leaving room for the vertical window-box boundary on the right, at line 2143 of my PR would be:
              “%-10.10s %-18.18s  %-20.20s %10.3f GiB”
which would replace the existing one:
              "%-10s %-25s  %-20s %10.3f GiB"
I’ll change this later.  Still have the “doesn’t pass formatting” issue to resolve.

Screenshot of 80-char wide current
From: PartialVolume <notifications@github.com>
Sent: Monday, February 10, 2020 02:57
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@mdcato<https://github.com/mdcato> I'm referring to before the wipe starts where there is slightly less columns due to the square brace selection [ ] being present. The disc size is then lost.

After the wipe starts the braces aren't there so more room for the rest text and the size is available and you don't see the problem.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PSKK4XC5TC3DLJOEJDRCEJETA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELHWXVA#issuecomment-584018900>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PRDBCU5ZG4J2BZYYBTRCEJETANCNFSM4KRHR44Q>.

mdcato commented:
@iderr,
Thanks for the info!

From: iderr <notifications@github.com>
Sent: Monday, February 10, 2020 12:30
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


Hello,
you take a look at this : https://github.com/IDerr/shredos
I took the old shreos package and repackaged it on a newer version of buildroot. Not tested yet, will do during this week, but it should work

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PS4LK5Z5STB5VRSKHLRCGMLTA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELJWIWY#issuecomment-584279131>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PUSOK6WPIWIVVNN6SDRCGMLTANCNFSM4KRHR44Q>.

PartialVolume commented:
@mdcato I'm going to shortly bump nwipe from 0.27rc1 to 0.27 for @martijnvanbrummelen to upload to debian sid this month. I don't want to add anything further into the master until we jump to 0.28rc2 which will be in a few weeks. I'm therefore going to leave this PR here while we work out the issues.

mdcato commented:
Sounds reasonable to me.


(sent from mobile)

________________________________
From: PartialVolume <notifications@github.com>
Sent: Tuesday, February 11, 2020 12:20:02 PM
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@mdcato<https://github.com/mdcato> I'm going to shortly bump nwipe from 0.27rc1 to 0.27 for @martijnvanbrummelen<https://github.com/martijnvanbrummelen> to upload to debian sid this month. I don't want to add anything further into the master until we jump to 0.28rc2 which will be in a few weeks. I'm therefore going to leave this PR here while we work out the issues.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PQDM4LWYWD6KHTEU2TRCLT5FA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELNQFDA#issuecomment-584778380>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PSTUPQPMF7DSIGYD73RCLT5FANCNFSM4KRHR44Q>.

mdcato commented:
I've canceled the Pull Request #193 and deleted the fork.

From: Mike Cato / Hays Technical Services <mike@haystech.com>
Sent: Tuesday, February 11, 2020 12:56
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>; martijnvanbrummelen/nwipe <reply@reply.github.com>
Cc: Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)

Sounds reasonable to me.

(sent from mobile)

________________________________
From: PartialVolume <notifications@github.com<mailto:notifications@github.com>>
Sent: Tuesday, February 11, 2020 12:20:02 PM
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com<mailto:nwipe@noreply.github.com>>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com<mailto:mike@haystech.com>>; Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


@mdcato<https://github.com/mdcato> I'm going to shortly bump nwipe from 0.27rc1 to 0.27 for @martijnvanbrummelen<https://github.com/martijnvanbrummelen> to upload to debian sid this month. I don't want to add anything further into the master until we jump to 0.28rc2 which will be in a few weeks. I'm therefore going to leave this PR here while we work out the issues.

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PQDM4LWYWD6KHTEU2TRCLT5FA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELNQFDA#issuecomment-584778380>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PSTUPQPMF7DSIGYD73RCLT5FANCNFSM4KRHR44Q>.

PartialVolume commented:
Thanks @mdcato , if you are redoing it you may want to split the America Vs US fix and the summary throughput I.e 0BS fix in their own separate branch/patch, makes it quicker to review and merge those as they are obvious fixes.

mdcato commented:
Yes, good suggestion.  I thought the American/US was (and it is) a small thing, but the clang-format v9.0.0 on Fedora 31 caused an unwanted wander into a rabbit hole.  Hopefully @louib’s fix will make it easier to update .clang-format file to work consistently between the various distributions.

From: PartialVolume <notifications@github.com>
Sent: Tuesday, February 11, 2020 15:48
To: martijnvanbrummelen/nwipe <nwipe@noreply.github.com>
Cc: Mike Cato / Hays Technical Services <mike@haystech.com>; Mention <mention@noreply.github.com>
Subject: Re: [martijnvanbrummelen/nwipe] 0.27rc1 proposed changes to GUI and throughput (#193)


Thanks @mdcato<https://github.com/mdcato> , if you are redoing it you may want to split the America Vs US fix and the summary throughput I.e 0BS fix in their own separate branch/patch, makes it quicker to review and merge those as they are obvious fixes.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/martijnvanbrummelen/nwipe/pull/193?email_source=notifications&email_token=ANGK2PXNMWKXF32JWLI3YA3RCMMKPA5CNFSM4KRHR442YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOELOGSUY#issuecomment-584870227>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ANGK2PSPWP45UWYA4SX4MV3RCMMKPANCNFSM4KRHR44Q>.
